helix-view/src/document.rs

```diff
@@ -112,11 +129,59 @@ pub fn canonicalize_path(path: &Path) -> std::io::Result<PathBuf> {
     std::env::current_dir().map(|current_dir| normalize_path(&current_dir.join(path)))
 }
 
+pub fn auto_detect_line_ending(doc: &Rope) -> LineEnding {
```

Comment:
We can probably move some of these issues into `helix_core::line_ending` (make a new module)